src/site/content/en/blog/module-workers/index.md

```diff
@@ -0,0 +1,232 @@
+worker, etc.), and all future imports reference the already-executed module instance. The loading
+and execution of JavaScript Modules is also optimized by browsers because a module's dependencies
+can be loaded prior to the module being executed, entire module trees can be loaded in parallel.
+Module loading also caches parsed code, which means modules that are used on the main thread and in
+a worker only need to be parsed once.
+
+Moving to JavaScript Modules also enables the use of [dynamic
+import](https://v8.dev/features/dynamic-import) for lazy-loading code without blocking execution of
+the worker. Dynamic import is much more explicit than using `importScripts()` to load dependencies,
+since the imported module's exports are returned rather than relying on global variables.
+
+##### worker.js:
+
+```js
+import { sayHello } from './greet.js';
+addEventListener('message', e => {
+  postMessage(sayHello());
+});
+```
+
+##### greet.js:
+
+```js
+import greetings from './data.js';
+export function sayHello() {
+  return greetings.hello;
+}
+```
+
+To ensure great performance, the old `importScripts` API is not available within module workers.
+Switching workers to use JavaScript Modules means all code is loaded in [strict
+mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode). Another
+notable change is that the value of `this` in the top-level scope of a JavaScript Module is
+`undefined`, whereas in classic workers the value is the worker's global scope. Fortunately, there
+has always been a `self` global that provides a reference to the global scope. It's available in
+all types of workers including service workers, as well as in the DOM.
+
+{% Aside %}
+Module workers also remove support for HTML-style comments.
+{% endAside %}
+
+## Preload workers with modulepreload
+
+One substantial performance improvement that comes with module workers is the ability to preload
+workers and their dependencies. With module workers, scripts are loaded and executed as standard
+JavaScript Modules, which means they can be preloaded and even pre-parsed using `modulepreload`:
+
+```html
+<!-- preloads worker.js and its dependencies: -->
+<link rel="modulepreload" href="worker.js">
+
+<script>
+  addEventListener('load', () => {
+    // our worker code is likely already parsed and ready to execute!
+    const worker = new Worker('worker.js', { type: 'module' });
+  });
+</script>
+```
+
+Preloaded modules can also be used by both the main thread and module workers. This is useful for
+modules that are imported in both contexts, or in cases where it's not possible to know in advance
+whether a module will be used on the main thread or in a worker.
+
+Previously, the options available for preloading web worker scripts were limited and not
+necessarily reliable. Classic workers had their own "worker" resource type for preloading, but no
```

Comment:
@jpmedley this was the past-tense paragraph I was lamenting. We're not removing classic workers, we're just adding a new modern alternative to them.

Reply:
'Classic workers had their own' --> 'Classic workers once had their own'

Reply:
@developit I think you have already made it clear that you're not removing classic workers because this is how you introduce the feature:

> A new mode for web workers with the ergonomics and performance benefits of [JavaScript
modules](https://v8.dev/features/modules) is shipping in Chrome 80, called module workers. 

`mode` gets the idea across clearly.